hpx/parallel/segmented_algorithms/transform.hpp

```diff
@@ -0,0 +1,876 @@
+                beg2 = traits2::begin(sit2);
+                end1 = traits1::local(last1);
+                if (beg1 != end1)
+                {
+                    out = dispatch(traits1::get_id(sit1), algo, policy,
+                        std::true_type(), beg1, end1, beg2, ldest,
+                        f, proj1, proj2);
+                }
+                last1 = traits1::compose(send1, hpx::util::get<0> (out));
+                last2 = traits2::compose(send2, hpx::util::get<1> (out));
+                dest = traits3::compose(sdest, hpx::util::get<2> (out));
+            }
+            return result::get(hpx::util::make_tuple<InIter1, InIter2, OutIter>
+                (std::move(last1), std::move(last2), std::move(dest)));
+        }
+
+        // parallel remote implementation
+        template <typename Algo, typename ExPolicy, typename InIter1,
+            typename InIter2, typename OutIter, typename F,
+            typename Proj1, typename Proj2>
+        static typename util::detail::algorithm_result<ExPolicy,
+            hpx::util::tuple<InIter1,InIter2,OutIter>
+        >::type
+        segmented_transform(Algo && algo, ExPolicy const& policy,
+            InIter1 first1, InIter1 last1, InIter2 first2, OutIter dest,
+            F && f, Proj1 && proj1, Proj2 && proj2, std::false_type)
+        {
+            typedef hpx::traits::segmented_iterator_traits<InIter1> traits1;
+            typedef hpx::traits::segmented_iterator_traits<InIter2> traits2;
+            typedef hpx::traits::segmented_iterator_traits<OutIter> traits3;
+            typedef typename traits1::segment_iterator segment_iterator1;
+            typedef typename traits1::local_iterator local_iterator_type1;
+            typedef typename traits2::segment_iterator segment_iterator2;
+            typedef typename traits2::local_iterator local_iterator_type2;
+            typedef typename traits3::segment_iterator segment_iterator3;
+            typedef typename traits3::local_iterator local_iterator_type3;
+            typedef util::detail::algorithm_result<ExPolicy,
+                hpx::util::tuple<InIter1,InIter2,OutIter>
+            > result;
+            typedef std::integral_constant<bool,
+                    !hpx::traits::is_forward_iterator<InIter1>::value ||
+                    !hpx::traits::is_forward_iterator<InIter2>::value
+                > forced_seq;
+            auto last2 = first2;
+            detail::advance(last2,std::size_t(last1 - first1));
+            segment_iterator1 sit1 = traits1::segment(first1);
+            segment_iterator1 send1 = traits1::segment(last1);
+            segment_iterator2 sit2 = traits2::segment(first2);
+            segment_iterator2 send2 = traits2::segment(last2);
+            segment_iterator3 sdest =  traits3::segment(dest);
+
+            std::vector<future<hpx::util::tuple<local_iterator_type1,
+             local_iterator_type2, local_iterator_type3> > > segments;
```

Comment:
@ajaivgeorge Can you improve the indentation here? A typedef for tuple may help here, you could reuse it at line 479.